core/arch/arm/include/kernel/secure_partition.h

```diff
@@ -0,0 +1,151 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (c) 2019, Linaro Limited
+ */
+
+#ifndef __KERNEL_SECURE_PARTITION_H
+#define __KERNEL_SECURE_PARTITION_H
+
+#include <assert.h>
+#include <kernel/tee_ta_manager.h>
+#include <kernel/thread.h>
+#include <kernel/user_mode_ctx_struct.h>
+#include <types_ext.h>
+#include <util.h>
+
+#define SP_RET_SUCCESS		0
+#define SP_RET_NOT_SUPPORTED	-1
+#define SP_RET_INVALID_PARAM	-2
+#define SP_RET_DENIED		-3
+#define SP_RET_NO_MEM		-5
+
+#define SP_VER_MAJOR_MASK	0x7FFF
+#define SP_VER_MINOR_MASK	0xFFFF
+#define SP_VER_MAJOR		(0 & SP_VER_MAJOR_MASK)
+#define SP_VER_MINOR		(1 & SP_VER_MINOR_MASK)
+#define SP_VERSION		((SP_VER_MAJOR << 16) | (SP_VER_MINOR))
+
+#define SP_MEM_ATTR_ACCESS_MASK		0x3
+#define SP_MEM_ATTR_ACCESS_NONE		0
+#define SP_MEM_ATTR_ACCESS_RW		1
+#define SP_MEM_ATTR_ACCESS_RO		3
+#define SP_MEM_ATTR_EXEC_NEVER		BIT(2)
+#define SP_MEM_ATTR_EXEC		0
+
+/* Used for EDK2 StMM communication */
+#define FFA_SVC_MEMORY_ATTRIBUTES_GET_64	UINT32_C(0xC4000064)
+#define FFA_SVC_MEMORY_ATTRIBUTES_SET_64	UINT32_C(0xC4000065)
+/*
+ * We need to define the RPMB IDs formally, since the plan is
+ * for them to be included in the FFA spec (for SP-to-SP future communication).
+ * This is fine for now as it represents the internal contract between the
+ * EDK2 RPMB driver and Secure Partition
+ */
+#define FFA_SVC_RPMB_READ		UINT32_C(0xC4000066)
+#define FFA_SVC_RPMB_WRITE		UINT32_C(0xC4000067)
+
+/* Param header types */
+#define SP_PARAM_EP			UINT8_C(0x01)
+#define SP_PARAM_IMAGE_BINARY		UINT8_C(0x02)
+#define SP_PARAM_BL31			UINT8_C(0x03)
+#define SP_PARAM_BL_LOAD_INFO		UINT8_C(0x04)
+#define SP_PARAM_BL_PARAMS		UINT8_C(0x05)
+#define SP_PARAM_PSCI_LIB_ARGS		UINT8_C(0x06)
+#define SP_PARAM_SP_IMAGE_BOOT_INFO	UINT8_C(0x07)
+
+/* Param header version */
+#define SP_PARAM_VERSION_1		UINT8_C(0x01)
+#define SP_PARAM_VERSION_2		UINT8_C(0x02)
+
+/* void syscall_sys_return(uint32_t ret) __noreturn; */
```

Comment:
These syscall defines doesn't seem to be used.

Reply:
None of those are, and only SP_PARAM_SP_IMAGE_BOOT_INFO is used from param header types. 
Shall I keep the types for future use and remove the stale syscall defines?